### ISSUE TITLE
unexpected "Escape" key behaviours

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-desktop#4128]
the "Escape" key is working only in some situations, and not always predictable.

some inconsistencies:

- [x] settings -> esc works
- [x] settings -> chats and media -> esc does not work (there is a dedicated issue for that at https://github.com/deltachat/deltachat-desktop/issues/3868)
- [x] settings -> chats and media -> back -> esc does not work
- [x] enlarge image -> esc works
- [ ] enlarge image -> context menu -> esc closes image and keeps context menu
- [x] enlarge image -> context menu -> close context menu -> esc does not work

EDIT: some more:

- [x] chat -> context menu -> esc works
- [ ] chat -> reaction menu -> esc does not work

**Comment by WofWca** [comment]
FYI some of these might get resolved in the Blueprint removal MR.

**Comment by maxphilippov** [comment]
Turns actually none of these are solved by #4006 cause a lot of the dialogs are actually implemented as a switch between different states rather than stacking, but this looks like a pretty straightforward refactor, I'll probably do that after Blueprint removal PR is merged since there's a lot of Dialog-related code

**Comment by maxphilippov** [comment]
> * enlarge image -> context menu -> esc closes image and keeps context menu

Just as an update for someone who's gonna work on this. On current `main` after Blueprint removal:
* enlarge image -> context manu -> esc closes both image and menu
Should close context menu first, I believe, so this is still a bug, just different behaviour

> * enlarge image -> context menu -> close context menu -> esc does not work

Escape works now

> * settings -> esc works
> * settings -> chats and media -> esc does not work (there is a dedicated issue for that at [Settings should be closable in sub-pages with escape key #3868](https://github.com/deltachat/deltachat-desktop/issues/3868))
> * settings -> chats and media -> back -> esc does not work

These are fixed in #4205

**Comment by WofWca** [comment]
As I said in https://github.com/deltachat/deltachat-desktop/pull/5191#pullrequestreview-2899513666, this is also the case in "New Chat" dialog, where contacts have context menu.

A more generic solution is preferable. For example, utilizing Popover and dialog API. For example, see how closing the "add members" dialog that is on top of "Create Group" dialog doesn't close the "create group" dialog.

**Comment by WofWca** [comment]
Let's reopen, for

> this is also the case in "New Chat" dialog

**Comment by ell1e** [comment]
I'll put this here and close the other issue afterward: opening a group's member list, then clicking an entry in a group's member list to see the profile, then pressing escape doesn't get me back to the member list, but closes everything, which is unexpected.

\<Edit by @WofWca\>: https://github.com/deltachat/deltachat-desktop/pull/6394.

**Comment by WofWca** [comment]
I think the proper and simplest way to fix this is to have separate `<dialog>`s for each view.

**Comment by WofWca** [comment]
More problematic places:
- [x] clicking on a contact in "View Group" dialog enters "view contact mode":
  https://github.com/deltachat/deltachat-desktop/blob/588020aa74b8c97b7240801b6e291693b1494c71/packages/frontend/src/components/dialogs/ViewGroup.tsx#L480

  https://github.com/deltachat/deltachat-desktop/pull/6394
- [x] "New Chat" dialog, where clicking "New Group" doesn't open a new dialog, but modifies the "New Chat" dialog.
  https://github.com/deltachat/deltachat-desktop/blob/1262cbdb6175e53eacbdec003e73d4021838fe33/packages/frontend/src/components/dialogs/CreateChat/index.tsx#L89

  https://github.com/deltachat/deltachat-desktop/pull/6405
- [ ] Just search for places where we utilize the "back" button:
  - [x] `onBack`
  - [ ] `onClickBack`
  - [ ] `<BackButton`.
- [ ] New Chat -> New Contact -> Escape (see _comment_ https://github.com/deltachat/deltachat-desktop/pull/6405#pullrequestreview-4484209394).

**Comment by omidz4t** [comment]
Following the Esc key improvements in #4128, I suggest that when the desktop app is in a narrow window size (mobile/compact view), pressing the Escape key should close the current chat and return the user to the chat list.

Currently, in this view, there is no quick way to go back to the messages list using the keyboard. Making Esc trigger the "back" action would ensure a consistent navigation experience across the app.

- [x] related pull req -> https://github.com/deltachat/deltachat-desktop/pull/5905

**Comment by r10s** [comment]
as a side note, when anyway on that, maybe it is easy to close the chat as well on esc, see feature proposal at https://support.delta.chat/t/closing-chats-with-escape-key-show-contact-list-hide-chat-log/5068 - but for now i would not make a dedicated issue for that. if it is not easy for whatever reason, let's leave it as a feature proposal for now and concentrate on more pressing issues

**Comment by nicodh** [comment]
@WofWca can we close this now?

**Comment by WofWca** [comment]
I edited my last comment. New Chat -> New Contact -> Escape.

**Comment by nicodh** [comment]
> I edited my last comment. New Chat -> New Contact -> Escape.

If this is the only issue left then I prefer close this and maybe open a new one for clarity. This issue is almost 2 years old and had so many changes ...

**Comment by WofWca** [comment]
- [ ] Add Profile -> Create New Profile -> Escape. The dialog will close instead of going back. Then if you click "Add Profile" again, the dialog will resume from the previous state.
